File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03042/p03042_num2_parsed.cpp
#include <iostream>
using namespace std;

int main() {
    string S;
    cin >> S;

    bool isYYMM = false;
    bool isMMYY = false;

    int firstTwoDigits = stoi(S.substr(0, 2));
    int lastTwoDigits = stoi(S.substr(2, 2));

    if (firstTwoDigits >= 1 && firstTwoDigits <= 12) {
        isMMYY = true;
    }
    if (lastTwoDigits >= 1 && lastTwoDigits <= 12) {
        isYYMM = true;
    }

    if (isYYMM && isMMYY) {
        cout << "AMBIGUOUS";
    } else if (isYYMM) {
        cout << "YYMM";
    } else if (isMMYY) {
        cout << "MMYY";
    } else {
        cout << "NA";
    }

    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03042/p03042_num2_parsed.py
S = input()

# Check for YYMM format
if int(S[:2]) >= 1 and int(S[:2]) <= 12:
    if int(S[2:]) >= 1 and int(S[2:]) <= 12:
        print('AMBIGUOUS')
    else:
        print('MMYY')
# Check for MMYY format
elif int(S[2:]) >= 1 and int(S[2:]) <= 12:
    print('YYMM')
else:
    print('NA')